### DISCUSSION TITLE
Zooz 800 series Zen 15 Not Reporting kWh Accurately

### DISCUSSION THREAD

**Discussion by bgreet** [zwave-js/zwave-js#7111] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
Zwave JS UI to home assistant is reporting a falsely elevated kWh report.  It records a device using much more kWh than it should and this reporting jumps over 5 minutes instead of what would normally be seen with a slow upward curve.

I have 8 Zen15 800 series and it seems to affect all of these devices.  Older Zen15 on 700 series chip does not seem to be affected (two are on my network).

What did you expect to happen instead?
Accurate data reporting

  1. Include Zen 15 800 series with S2 security
  2. Update to latest firmware (though also happens with older firmware)
  3. Plug in device and monitor kWh over time

Anything else we should know?

Please see below picture examples of reporting on 8/12/24:

Utility output room zen15 inaccuracy occurred at 725pm to 730pm.  It is node 002 in zwavejs UI.  800 series zen15 Firmware 2.10

![image](https://github.com/user-attachments/assets/6ecc60ce-554c-469e-8043-4ddacb91b952)

Electric Freezer zen15 is node 054.  Inaccuracies occurred sampling between 08:25am and 08:30am and 8:05 to 8:10pm. 800 series zen15 Firmware 2.10

![image](https://github.com/user-attachments/assets/49376589-7aaa-4512-83b5-55e8d2459622)

3D printer switch is node 014.  Error in reporting occurred between 0725 and 0730am.  800 series zen15 Firmware 2.0

![image](https://github.com/user-attachments/assets/a1b5bfcc-16c1-45f2-aa61-0b8fadaf2e31)

### Software versions

ZWAVE JS UI run as docker container, Home Assistant run as docker container

zwave-js-ui: 9.17.0.4f4bb9d
zwave-js: 13.1.0
Z wave controller: Zooz ZSET39 LR Firmware 1.40

### Device information

Manufacturer: Zooz
Model name: Zen15 800 series
Node ID: 
002
003
013
014
054
074
075
091

The device nodes given in this example, however, are nodes 002, 014 and 054 

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-08-12.zip](https://github.com/user-attachments/files/16620929/zwavejs_2024-08-12.zip)

**Comment by AlCalzone** [comment]
You'll have to make a different log if we want to prove this is a device issue. The problem is that the encrypted payload is never logged in plaintext, so decoding the meter reading by hand isn't possible. To do so, execute the following [driver function](https://zwave-js.github.io/zwave-js-ui/#/usage/driver_function?id=examples):
```ts
process.env.NODE_ENV = "development"
```

This will enable plaintext logging of several encryption-related things. If you look at the written logfile, you should see the plaintext being logged, similar to this:
```
└─[Security2CCMessageEncapsulation]
  │ sequence number: 79
  │ security class:  S2_Authenticated
  │ plaintext:       0x...
```

Afterwards, keep Z-UI running until you've captured such an event again. Once you're done, unset the env variable again:
```ts
process.env.NODE_ENV = ""
```

You may want to send me that logfile privately, so I can remove sensitive information before uploading it.

**Comment by bgreet** [comment]
Thank you so much for looking into this.  Will look to get you this tonight
when I get home from work

**Comment by dualtronic** [comment]
Bump. This is still an active problem (test on zen15 ver 2.1 with 800 series chip). Zooz isn't responding to my support request.  I don't know if they are stalling or expecting some type of evidence that zwave-js isn't the cause.

**Reply by AlCalzone** [reply]
I don't recall ever having received a log like I requested in my previous comment.

**Comment by peterfares** [comment]
I also have this issue with ZEN15 800
I have two of the devices, both initially on 2.0.1
Only one of them has this issue, where it randomly jumps up. I tried updating it to 2.10.0 and it is still having the issue. 
Seems like it might be an issue with the device itself though, doesn't the Energy Consumption number come directly from the device?

**Reply by AlCalzone** [reply]
It does

**Comment by dualtronic** [comment]
Could the Electric Consumption [kWh] error be related to an incorrect PropertyKey value?  I can't find much information about PropertyKey, but it appears to be some kind of binary mask?  The dubug Info window shows PropertyKey is 65537 (binary: 00010000000000000001).  Could the higher order bit be causing trouble?

**Reply by AlCalzone** [reply]
No that's just how Z-Wave JS distinguishes values internally.

**Comment by hceuterpe** [comment]
So I wonder if this is related to the new params added with the ZEN15 800LR.  

There's a pull request with the new added params.  They deal with energy monitoring threshold and frequency.  Params 173, 174, 175, 176 are all set to 0 for me (0 is default for 173 and 174), for the rest my home has really stable power so not too worried about voltage readings for every power metering switch, so I only go with watts and kWh to cut down on z-wave traffic.  I get rather consistent graphs, unlike yours, and they are connected to space heaters so they are consuming a good amount of juice when on.

My guess is my settings are such that it reports kWh every 5 minutes in fixed intervals, whereas for you the kWh threshold somehow is overriding the reporting intervals entirely.

![Screenshot (170)](https://github.com/user-attachments/assets/d41b93bb-dc6c-4a5d-a7fd-54c18dfe17ea)
![Screenshot (171)](https://github.com/user-attachments/assets/21937ae6-5883-4678-86e2-b16278a5eaf3)
![Screenshot (172)](https://github.com/user-attachments/assets/c3665ebe-f5a5-425e-a870-8e9ac452877f)

**Reply by dualtronic** [reply]
@hceuterpe 
It's possible, but this would point the blame back to the zooz firmware, right? If we could provide a specific set of parameters that turn the problem on and off, then Zooz must do another review.  Currently, their help desk is convinced that it's only a HA problem and they're not doing anything else to resolve the issue.  I suspect every customer has this same problem, but only HA users bother to report it.

**Reply by AlCalzone** [reply]
Just as a reminder - I outlined [here](https://github.com/zwave-js/node-zwave-js/discussions/7111#discussioncomment-10344835) how to generate logs that can prove the device is reporting incorrect values. If you folks can reproduce reliably, that could help show Zooz what to look for.

**Reply by hceuterpe** [reply]
> @hceuterpe It's possible, but this would point the blame back to the zooz firmware, right? If we could provide a specific set of parameters that turn the problem on and off, then Zooz must do another review. Currently, their help desk is convinced that it's only a HA problem and they're not doing anything else to resolve the issue. I suspect every customer has this same problem, but only HA users bother to report it.

I suggest trying the updated config file, and settings modifications to see if it works for you? I've seen bugs impacting other Z-wave devices where something like a power metering threshold config setting would completely override its otherwise related reporting interval config, such that the device would only report with the former and completely ignore the latter.  Given KWh is a cumulative value instead of one that's a point in time the interval reporting at least imo would be far more useful.  Obviously for W/V/A, having both is useful, and at least for watts the threshold and interval together seems to still work for me...

**Reply by dualtronic** [reply]
@AlCalzone The error may not occur for weeks but then happen several times in one day (like today).  The problem is so infrequent and unpredictable that it's impractical on my home's HA.

**Reply by SquaredCircled** [reply]
I have experienced the same behavior.
Home Assistant Yellow
Zooz ZAC73 controller
ZEN15 800
Z-Wave JS UI
S2 Encryption

Only one of my Zen15 relays exhibits the problem. When swapping two of them between different loads the faulty readings appear on the same device.

Zooz support also blamed HA

Has anyone seen the issue when not using encryption?

**Comment by tofurky** [comment]
I've noticed the same on my newly provisioned ZEN15 running the latest firmware (v2.10.0). I have an older ZEN15 (500 series I think) which didn't show the issue.

I've been running zwave-js-ui with NODEENV=debug, which is generating about 80MB of (uncompressed) logs per day for a week now (starting 2025/03/04). I was hoping that it'd do it again, but it has not. There seems to not be really any resource issues with the verbose logging enabled so I'll just leave it like this for the time being.

![image](https://github.com/user-attachments/assets/8b21b7ae-16b2-4c03-8ba5-30a5f1283af8)

I noticed this because I had negative power draw in the Energy dashboard versus what utility was showing as consumed.

![image](https://github.com/user-attachments/assets/ab208af2-25c9-4052-ab34-d3fe3eb5344f)

On another Zooz product (ZEN25 double plug) I'd see wild spikes in instantaneous wattage (like, 10kW on a < 10W load) but I wasn't paying any attention to kWh at that time. Opened a support case a few years ago and they swapped it out, but it made no difference. I think this is a different issue (W vs kWh) but it sort of makes me think this isn't the first buggy product they've shipped. An aside, the ZEN25 just repeatedly clicks its relays now (bricked itself) when plugged in after it sat on the shelf for 6mo or so.

I'll also see very sporadic wild W readings on my ancient Aeon Labs DSC06 plugs, but I'm almost positive it's due to corrupt packets. Those don't even support Z-Wave+ (i.e. proper checksums). Sometimes random entities for the wrong class of product shows up in that case as well.

**Comment by tofurky** [comment]
![image](https://github.com/user-attachments/assets/0584d344-7efa-42cd-b116-216c1fb39e8f)

Finally it happened, after checking more or less daily for over a month. Sudden jump from 122 -> 328 kWh.

```
2025-04-14T01:34:55.675Z DRIVER « [Node 025] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -83 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 68
                                    │ extensions:
                                    │ · type: SPAN  sender EI: 0xsanitized
                                    │ key:             0xsanitized
                                    │ IV:              0xsanitized
                                    │ plaintext:       0xsanitized
                                    │ auth data:       0xsanitized
                                    │ auth tag:        0xsanitized
                                    │ security class:  S2_Authenticated
                                    └─[SupervisionCCGet]
                                      │ session id:      23
                                      │ request updates: false
                                      └─[MeterCCReport]
                                          meter type:  Electric
                                          scale:       kWh
                                          rate type:   Consumed
                                          value:       122.657
                                          time delta:  300 seconds
                                          prev. value: 122.657

2025-04-14T01:40:19.579Z DRIVER « [Node 025] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -83 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 166
                                    │ key:             0xsanitized
                                    │ IV:              0xsanitized
                                    │ plaintext:       0xsanitized
                                    │ auth data:       0xsanitized
                                    │ auth tag:        0xsanitized
                                    │ security class:  S2_Authenticated
                                    └─[SupervisionCCGet]
                                      │ session id:      43
                                      │ request updates: false
                                      └─[MeterCCReport]
                                          meter type:  Electric
                                          scale:       kWh
                                          rate type:   Consumed
                                          value:       348.516
                                          prev. value: 122.657
```

Could send full log (e-mail on git commit...?) if needed.

**Reply by AlCalzone** [reply]
Yes, please! The email in my git commits is fine.

**Reply by tofurky** [reply]
> Yes, please! The email in my git commits is fine.

Done, thanks.

**Reply by AlCalzone** [reply]
I've sent Zooz an email with what I could gather from that log.

**Reply by dualtronic** [reply]
@AlCalzone Any news from Zooz?  Hopefully they are taking this seriously and not blaming their customers again.

**Reply by AlCalzone** [reply]
Yeah, they have a suspicion and are in contact with their engineering.

**Reply by PaddyDHurley** [reply]
@AlCalzone the zen05 does better than the zen15.   I was counciled by the zooz technical support that I should not be using the zen05 for my 1/3 horsepower 9.5 amp sump pump.   I switched to the zen15 and it reports high voltage, incorrect amps compared to the zen05 which always reported 9.5 amps.  During a recent rain the zen15 reported 10 amps on one start and 5 amps on the next.   The pump always pulls 9.5 amps as expected on the zen05.  

The zen15 needs to be fixed by Zooz but I cannot find a replacement.   I am considering going back to the zen05.

**Comment by xuexiaofei** [comment]
![1746504071313](https://github.com/user-attachments/assets/f417e43b-f931-4a4d-b865-b8175f02b2a4)
@tofurky The time delta field is missing here. Did you omit it, or was it not present in the original log? Could you please provide the complete log?

**Reply by AlCalzone** [reply]
@xuexiaofei I can help with that.

The last good reading is the following:
```
2025-04-14T01:34:55.675Z DRIVER « [Node 025] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -83 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 68
                                    │ plaintext:       0x6c01170e320221640001df21012c0001df21
                                    │ security class:  S2_Authenticated
                                    └─[SupervisionCCGet]
                                      │ session id:      23
                                      │ request updates: false
                                      └─[MeterCCReport]
                                          meter type:  Electric
                                          scale:       kWh
                                          rate type:   Consumed
                                          value:       122.657
                                          time delta:  300 seconds
                                          prev. value: 122.657
```

I've also decoded the plaintext so you can see for yourself:
```
6c Supervision CC
01 Get
17 Session ID 23
0e 14 bytes encapsulated
32 Meter CC
02 Report
21 =0_01_00001 - Type 1 (Electric), Consumed
64 =011_00_100 - Precision 3, Scale 0 (kWh), Size 4
0001df21 122657 - current value
012c Time delta 300s
0001df21 122657 - previous value
```

The next reading has a huge jump in kWh:
```
2025-04-14T01:40:19.579Z DRIVER « [Node 025] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -83 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 166
                                    │ plaintext:       0x6c012b0e3202216400055164ffff0001df21
                                    │ security class:  S2_Authenticated
                                    └─[SupervisionCCGet]
                                      │ session id:      43
                                      │ request updates: false
                                      └─[MeterCCReport]
                                          meter type:  Electric
                                          scale:       kWh
                                          rate type:   Consumed
                                          value:       348.516
                                          prev. value: 122.657
```

Again with decoded plaintext:
```
6c Supervision CC
01 Get
2b Session ID 43
0e 14 bytes encapsulated
32 Meter CC
02 Report
21 =0_01_00001 - Type 1 (Electric), Consumed
64 =011_00_100 - Precision 3, Scale 0 (kWh), Size 4
00055164 348516 - current value
ffff - unknown time elapsed
0001df21 122657 - previous value
```

Notice that the elapsed time is encoded as unknown, which is why Z-Wave JS doesn't log it.

**Reply by xuexiaofei** [reply]
OK, thank you, I will try to analyze and solve this problem.

**Comment by luckyjay** [comment]
Found this thread today after noticing the same issue with a ZEN15 I recently added. I have others that have no issue. But the recent one was added with security and it's the newer ZEN15 800LR. Added in regular zwave mode (not LR). kWh usage goes up faster than my entire house! Load is moderate, around 400-600 watts, there's a mobile AC unit on it.

**Reply by luckyjay** [reply]
I just noticed that the device seems to reset right around the time this happens. The AC unit power cycles like it lost power. I'm not always near it so I didn't notice until a recent occurrence.

**Reply by bgreet** [reply]
> I just noticed that the device seems to reset right around the time this happens. The AC unit power cycles like it lost power. I'm not always near it so I didn't notice until a recent occurrence.

I've noticed the same exact behavior as well

**Reply by luckyjay** [reply]
I opened a ticket with Zooz to see what they say.

**Reply by bgreet** [reply]
When I did that they just blamed zwavejs

**Reply by dualtronic** [reply]
Yah, Zooz blames zwavejs and we blame Zooz.  They don't want to think about it anymore becuse they did a "full review" last year and decided their firmware is not to blame. However, they cannot explain how zwavejs could be doing this since the "jump" isn't temporary. It remains beyond power cycles. So that jump in the value has to be saved somewhere.  zwavejs just reads their number. It doesn't maintain some variable that offsets the value. If it did, then it would correct itself if we deleted the device and setup the same hardware again.  It's infuriating.

**Reply by luckyjay** [reply]
Indeed, I basically got the same response.

For the last two days it hasn't happened. Hmm.

**Reply by bgreet** [reply]
I used to love zooz but recently I've had a lot of issues.  Their zen51 after a firmware update months ago won't allow local control of a switch and they haven't fixed it.  They gaslighted me on the zen15 (though the 700 series seems to work okay) trying to convince me their is no issue.  I became worried when I would connect them to my computers and the computer would randomly turn off.

**Reply by tofurky** [reply]
ZEN15 is conveniently missing from the Works with HA announcement: https://www.home-assistant.io/blog/2025/07/15/zooz-joins-works-with-home-assistant/

I also have a bunch of ZEN04 (listed) which work fine.

Hopefully there's an updated firmware released that fixes the issue.

**Comment by dpifke** [comment]
I had reported this issue to Zooz back in January, and they just reached out to me today to let me know that a firmware update is available which may solve the problem:

> Hi there,
> 
> Thank you again for reporting the kWh reading issue with your device.
> 
> We've seen sporadic reports from Z-Wave JS users—particularly on the Home
> Assistant and HomeSeer platforms—where energy consumption suddenly spikes to
> unusually high values that don't match the actual load.
> 
> We want to clarify that this behavior has only been observed on the Z-Wave JS
> platform. After extensive internal testing, we haven’t seen the same issue
> occur on other hubs. That said, we’ve been working closely with the Z-Wave JS
> team and our own software engineers to better understand and address the
> issue.
> 
> We’re happy to share that we now have a firmware update available that may
> help resolve this behavior when used with Z-Wave JS. The new firmware is
> attached to this message, along with basic update instructions: [How to
> Perform an OTA Firmware Update on Zooz Devices][1]
> 
> The new firmware includes the following changes:
> 
>   * Minimized flash write operations by avoiding writes when the kWh value remains unchanged.
>   * Optimized the kWh calculation process to prevent overflow in pulse counting.
> 
> If you have any questions during the update process or continue to see
> irregular readings after applying the firmware, please don’t hesitate to reach
> out.
> 
> Sara

I just installed this on one of my ZEN15s which has pretty reliably been showing this issue, and will report back.

Edit: Zooz has given me permission to share the beta firmware here:

[ZEN15_V02R21P02_beta.zip](https://github.com/user-attachments/files/21323912/ZEN15_V02R21P02_beta.zip)

(To install, you'll need to extract the .gbl from the .zip archive, as Github allows uploading the latter but not the former.)

**Reply by mjohnson9** [reply]
I'm frustrated by the continued insinuations that this is a Z-Wave JS issue, absent any compelling evidence. In fact, the issues they say they've fixed in this email don't appear remotely unique to Z-Wave JS. It seems most likely to me that users of most other hubs either lack the ability to actually monitor this value or choose not to. I'm not sure how they didn't figure that out rather quickly. Zooz really dropped the ball with that, and I don't know if it's bad development practices or intellectual dishonesty in the pursuit of profit. Prior to experiencing this issue, I was a fan of theirs because of the wide selection of useful devices, and now I'm substantially more hesitant to trust them. I hope they're reading this and will choose to act in a way to rebuild trust.

Regardless, at least they've ended up finally doing something about this. Here's hoping this actually fixes the issue.

**Reply by bgreet** [reply]
Completely agree.  Seemed like the easy thing was to say it was just zwave js as opposed to trying to actually figure out the issue. I've also now had issues with their zen51 dry contacts where the latest firmware disabled local control.

**Reply by dualtronic** [reply]
"We are proud to correct a firmware error that your software caused. Until now, we did not expend any serious effort because your platform does not provide customer revenue. And therefore, the error is not real to us. Lucky for you, the error was finally discovered on a platform that is valuable to upper management. This is why the problem you had was caused by your software. If your software does not exist to us, the problem doesn't either. However, now it is corrected for you after ignoring your support requests for years. You're welcome!"

**Reply by rrozema** [reply]
Not saying there is\was no problem in the firmware or that the problem can't be mitigated by a firmware change, but there are more reports on spikes showing in energy used readings for other devices, not just Zooz'. I f.e. sporadically experience similar issues for non Zooz z-wave devices, Zigbee devices, Plugwise devices and even for the PowerCalc integration (which is software-only, with no hardware nor firmware involved). Most often the problem seems to occur after a device was  (temporarily) not available or HA was restarted. It doesn't happen on all devices, not even on all devices of the same brand, model and make, but if it happens in one device that device is likely to experience it more than once. I have a lot of energy sensors in my system (almost every device in my home has an energy entity connected to it plus I have extensive reporting on them) so I am bound to notice such anomalies more than the average HA user.

In their defence, it feels to me like Zooz may have a point when they say it's not _caused_ by their firmware. And as such I don't agree with the negative attitude shown towards them here: they actually spent time on investigating it and did come up with a work around! A lot of other brands -including Plugwise, who generally do give very thorough support on most issues- do not report back on this issue at all.

My personal guess is that there is a very subtle initialisation problem somewhere at the core level of HA though, not in zwave JS. Plus that this gets triggered only rarely and in very specific circumstances which include some 'behaviour' of the device or the integration used.

**Reply by tofurky** [reply]
It's actually probably a memory corruption bug in the C firmware running on the ZEN15. Either the flash which keeps track of the value is corrupted, or there's some kind of overflow / mishandling of memory during runtime.

Some have reported the device power cycling when it happens (mine is in the crawlspace, so no idea) which is further evidence to that.

The debug logs above show the value is coming from the device itself.

I flashed the updated firmware; will of course report back if it happens again with it (opened a support case with Zooz as well, finally).

**Reply by mjohnson9** [reply]
I just updated one of my ZEN15 that I'm using specifically for energy monitoring with this beta firmware. It's monitoring a relatively high power device (a dehumidifier, which is fundamentally a reconfigured window air conditioner), which seemed like it might be part of what triggered this bug before. I'll see if I have anymore implausible 100+ kW spikes in the next few weeks.

**Reply by mjohnson9** [reply]
Well, immediately after update, I received an alert because the derived power from the meter sat at 5.5kW (+/- 100W continuous variance) for approximately 180 seconds (+/- 45 seconds because of sample rate). That's exceptionally implausible because this is a 20A circuit (nominal 120V). At that power, it would be drawing around 45A, which would nearly instantaneously trip the breaker (or set the wiring on fire if the breaker failed closed).

After the initial period, though, the derived power settled and began to match the reported instantaneous power within a reasonable margin.

**Reply by tofurky** [reply]
My ZEN15 jumped from 5724KWh to 6442KWh earlier today even with the beta firmware installed on it.

<img width="1106" height="370" alt="image" src="https://github.com/user-attachments/assets/1f87edb7-f4d5-42b3-a904-52d0dbcc8aac" />

**Reply by dpifke** [reply]
I upgraded three devices (monitoring my clothes washer, microwave, and refrigerator) to the beta firmware on the evening of July 18, and another (monitoring my dishwasher) on the evening of July 19.

The clothes washer jumped during (or right after) the upgrade.  The dishwasher jumped the day before it was upgraded, and then again during (or right after) the upgrade.  It had most consistently been showing the problem.

Since then, I haven't seen a jump.
<img width="1947" height="695" alt="image" src="https://github.com/user-attachments/assets/fd5a520c-09c1-4457-8276-4f511c8a1836" />

Edit: zooming in, I think the other two might have shown a jump right around the time they were upgraded, although not nearly as dramatic, and not since.

**Reply by luckyjay** [reply]
I'm still getting jumps

**Comment by tofurky** [comment]
I was sent a replacement device (not my idea/request; doesn't seem like bad HW to me) as this is apparently hard to reproduce and they wanted the original back; and, within 24 hours of being set up, I've had a sudden jump of several KWh on the replacement (using factory firmware, not the beta one):

<img width="565" height="556" alt="image" src="https://github.com/user-attachments/assets/920d084d-d353-4877-bfdb-6d6170ca7169" />

**Comment by AlCalzone** [comment]
Are y'all in contact with Zooz support or should I poke them about this again?

**Reply by tofurky** [reply]
In contact and received a replacement unit (which has the same issue), but they're still implying that even if it's a firmware bug on the device, it's something unique to zwave-js. That doesn't really make sense to me considering the device is responsible for updating/storing the accumulated KWh.

**Reply by hceuterpe** [reply]
> In contact and received a replacement unit (which has the same issue), but they're still implying that even if it's a firmware bug on the device, it's something unique to zwave-js. That doesn't really make sense to me considering the device is responsible for updating/storing the accumulated KWh.

This is what my metering shows.  No jumps on any.  I have 7 in total, 4 of the 800lr and 3 of the older 500 series. At least for me the 800lr switches actually seem to work a lot better than the older ones.  I have also included my configuration for my zen15 if you're interested in trying that.  It is different than default and I want to say I recall it working better after the changes.  I don't monitor voltage and amperage as my house gets pretty reliable power.    Mainly just watts and kWh.
![Screenshot_20250807_171632_Brave](https://github.com/user-attachments/assets/04c333e4-c6e1-4a6a-9aa0-47f0097745c6)
<img width="1440" height="1128" alt="Screenshot_20250807_171531_Home Assistant" src="https://github.com/user-attachments/assets/bde504f4-63c1-4893-a549-c3c27b6f8201" />

**Comment by luckyjay** [comment]
Today Zooz sent me newer beta firmware, 2.23.1. They apparently did more testing and reproduced the device reset that some of us have noticed corresponds to the sudden rise in kWh. They say the update "may help resolve this behavior when used with Z-Wave JS". Interesting, still saying there's some connection with using z-wave JS...

**Reply by dualtronic** [reply]
Zooz admitted that unspecified malformed Z-wave communication has authority to stop their device, corrupt its NVM, and reset its micro?  Doesn't matter what RF communication is doing, the device should never react in such a way.

**Reply by AlCalzone** [reply]
They insist that they have never seen this behavior with other drivers/hubs. I kinda doubt it, more likely that users from other hubs are less focused on stuff like energy measuring and don't notice this as easily. Or they are a much smaller part of their customer base. Who knows...

**Reply by dualtronic** [reply]
What I'm saying is that Zooz has a problem they must fix.  Z-wave communication (of any kind) must never have a way to corrupt local NVM.  At the very least, Zooz is responsible to ensure their NVM is safe during an unanticipated micro reset.  That's basic EMC checklist validation.

**Reply by luckyjay** [reply]
I've used this firmware with the appliance running for two days. So far, no resets or sudden kWh increase... but that's not long enough to know for sure because it could go that long before without an issue.

Something else I noticed is that the kWh reading went **way** down right after the new firmware. Previously at just over 29,000 kWh (completely wrong of course) it went down to 436 kWh, which sounds more realistic. And has been smoothly rising when used.

**Reply by dualtronic** [reply]
@luckyjay That sounds like a bit shift, or an illegal memory copy.  I hope it is corrected. Please keep us informed. Thanks!

**Reply by AlCalzone** [reply]
I've been told it was an unexpected reset of the MCU that deals with the power switching and usage estimation, but without resetting the usage estimator for whatever reason. So the Z-Wave chip continued to read wrong data from that chip, accumulating unrealistic usage stats.

Not related to Z-Wave traffic whatsoever.

**Reply by hceuterpe** [reply]
> Today Zooz sent me newer beta firmware, 2.23.1. They apparently did more testing and reproduced the device reset that some of us have noticed corresponds to the sudden rise in kWh. They say the update "may help resolve this behavior when used with Z-Wave JS". Interesting, still saying there's some connection with using z-wave JS...

Any chance you'd be able to share the newer beta firmware?

**Reply by juched78** [reply]
Please share. Thank you.

**Reply by AlCalzone** [reply]
Have you tried reaching out to Zooz support?

**Reply by juched78** [reply]
I just opened a ticket with them.

**Reply by juched78** [reply]
They responded quickly with attached firmware. Uploaded and running. Will see how it performs. 
[800_SwitchZEN15_US-LR_0x027A_V2.23.01_beta.gbl.zip](https://github.com/user-attachments/files/23139006/800_SwitchZEN15_US-LR_0x027A_V2.23.01_beta.gbl.zip)

**Reply by tofurky** [reply]
I've been running it since September 24th - no kWh jumps since. I let Zooz support know the same.

**Reply by rwpotter3rd** [reply]
Hello everyone. I am experiencing similar issues with some ZEN15 devices I got for Christmas and was wondering if this beta firmware seems like a good fix. Thanks in advance!

**Reply by tofurky** [reply]
Still no issues in the past few months w/said firmware.

**Reply by bgreet** [reply]
I'm still having issues with restarts just far less frequent.  I would still recommend not connecting to any sensitive electronics

**Reply by tofurky** [reply]
I suppose the plug might be randomly power cycling down the the crawlspace. I'm just going by the issue I can see from the HA side - the jumps in kWh, which do seem resolved.

**Reply by hceuterpe** [reply]
> I suppose the plug might be randomly power cycling down the the crawlspace. I'm just going by the issue I can see from the HA side - the jumps in kWh, which do seem resolved.

Have you tried the 2.30 firmware?

**Reply by tofurky** [reply]
> Have you tried the 2.30 firmware?

Not yet; wasn't aware there was a newer one. Mine is on the (unreleased?) v2.23.1 provided via their support. I imagine 2.30 includes the fix from that. However, 2.30 isn't showing on the changelog to describe what it does: https://www.support.getzooz.com/kb/article/309-zen15%C2%A0power-switch-change-log/

**Reply by bgreet** [reply]
Same, wasn't aware there is a 2.3 released (I'm on 2.23)

**Reply by tofurky** [reply]
Timestamp in the zip is November; I'm flashing it now. Will report issues if I encounter them.
```
$ unzip -l ZEN15_V02R30_US.zip 
Archive:  ZEN15_V02R30_US.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
   158760  2025-11-11 19:55   ZEN15_V02R30_US.gbl
---------                     -------
   158760                     1 file
```

**Reply by rwpotter3rd** [reply]
I found the 2.30 firmware just the other day after I commented initially. I flashed it onto 3 ZEN15 devices. All 3 have stopped reporting kWh jumps. 2 of the 3 also seem to have stopped the random power cycles - they've been good for a few days solid now. The 3rd one is power cycling about once every 24-36 hours. I just got super frustrated and unplugged it 10 minutes ago.

**Reply by tofurky** [reply]
Changelog was just updated (I've got a urlwatch job set up for these): https://www.support.getzooz.com/kb/article/309-zen15%C2%A0power-switch-change-log/

Firmware 2.30 (released 11/2025 as OTA)

- Fixed kWh Reporting Issue: Resolved an edge-case issue where ZEN15 could report inflated kWh values. The root cause was identified as the power metering chip not being properly cleared after an abnormal device restart.

- Updated Factory Reset Behavior: Improved the factory reset procedure to ensure kWh data is fully cleared. After holding the button for 20 seconds, ZEN15 will now restore default settings and reset accumulated kWh values.

- Optimized Power Report Frequency (Parameter 171): Changed the default value of Parameter 171 (Power Report Frequency) from 30 seconds to 300 seconds to reduce unnecessary traffic on the Z-Wave network.

    - Devices using the previous default value (30 seconds) will be automatically updated to 300 seconds.

    - Devices with a custom value configured will remain unchanged.

**Reply by dualtronic** [reply]
Therefore, every time I see a data blip, it had restarted abnormally.  That's troubling given the frequency of data blips I have.  It must be resetting 6-10 times a week for unknown reasons.

**Comment by bgreet** [comment]
I had to decommission these things (I bought like 5) because they would randomly turn off.  You could imagine that may not be the best thing when you are on a treadmill and the thing suddenly shuts off.

**Comment by PaddyDHurley** [comment]
The Zen05 reports accurately.  The ZEN15 is not a reliable reporter and their technical support is terrible.
